native/core/src/events/calendar.rs

```diff
@@ -56,6 +56,100 @@ pub enum EventLocation {
     },
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct EventLocationInfo {
+    pub location_type: String,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub description: Option<TextMessageEventContent>,
+    pub icon: Option<Icon>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub coordinates: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub uri: Option<String>,
```

Comment:
why copy this over rather than just have the `EventLocation` as an inner type? Why is this type needed at all actually?

Reply:
Good thinking, this is a nice improvement and will reduce redundancy. I'll change it! 👍🏻